docs/docs/how-to/plugins-and-themes/creating-a-source-plugin.md

```diff
@@ -806,6 +806,105 @@ One challenge when developing locally is that a developer might make modificatio
 
 Some data sources keep event logs and are able to return a list of objects modified since a given time. If you're building a source plugin, you can store the last time you fetched data using the [cache](/docs/creating-a-source-plugin/#caching-data-between-runs) and then only sync down nodes that have been modified since that time. [`gatsby-source-contentful`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-source-contentful) is an example of a source plugin that does this.
 
+### Enabling Content Sync
+
+If you would like to add Content Sync to your source plugin here but aren't sure what it is [learn more about Content Sync here](/docs/conceptual/content-sync/). To enable this feature in your source plugin you will need to make sure that your data source (or CMS) also works with Content Sync. The first thing you'll want to do is identify which nodes you'll want to create a node manifest for. These will typically be nodes that you can preview, entry nodes, top level nodes, etc. An example of this could be a blog post or an article, any node that can be the "owner" of a page. Once you've identified that you'll want to use the [unstable_createNodeManifest action](/docs/reference/config-files/actions/#unstable_createNodeManifest) to create these manifests. A good place to call this action is before you call `createNode`.
+
+An easy way to keep track of your manifest logic is to parse it out into a different util function. Either inside the `createNodeManifest` util or before you call it you'll need to vet which nodes you'll want to create manifests for.
+
+```javascript:title=source-plugin/gatsby-node.js
+import { createNodeManifest } from "./utils.js"
+exports.sourceNodes = async (
+  { actions }
+) => {
+  // sourcing data...
+  const { unstable_createNodeManifest, createNode } = actions
+
+  nodes.forEach(node => {
+    // highlight-start
+    const gatsbyNode = createNode(node)
+
+    const nodeIsEntryNode = `some condition`
+    if (nodeIsEntryNode) {
+      createNodeManifest({
+        entryItem: node,
+        entryNode: gatsbyNode,
+        project,
+        unstable_createNodeManifest,
+      })
+    }
+    // highlight-end
+  })
+}
+```
+
+There are a couple more checks we need to do inside the util. First, at the moment you'll only want to create node manifests for preview content, next we need to check if the Gatsby version supports `unstable_createNodeManifest`. Finally we'll want to parse out the time the content was updated and create a `manifestId` with all that information. The manifest id is the link between the data source and the Gatsby site. This manifest id will need to be created identically in the data source and here in your source plugin. Because of this we will use information given from the data source to build up the manifest id.
```

Comment:
I think this could be easier to parse out with subheadings for each thing needed to implement it.

Reply:
I like the idea of numbered steps to make it easier to follow, but the content is great 👍